pull_request_guidelines.md

```diff
@@ -0,0 +1,78 @@
+## Why do we do pull requests?
+
+* Catching and reducing technical debt
+* Catching mistakes
+* Improving development skills
+* Familiarity with the whole codebase
+* Maintaining consistency
+
+## Pull Request Software
+
+* Reviewboard vs. Github
+
+## Creating Pull Requests
+
+* Pull request granularity (how big should my pull requests be?)
+ * Try to keep them under 500 lines.
```

Comment:
It's probably a bit out of scope for this document, but it might be worthwhile to write up ways of rewriting history to separate multiple responsibilities that got accidentally lumped together.

In the same vein, we might want to address the right way to file multiple PRs at once (do you have one based off the other and then rebase onto master before the merge, or do you just let the magic green button handle it and get some slightly weird-looking history?)